src/OpenTelemetry/Logs/ActivityEventAttachingLogProcessor.cs

```diff
@@ -0,0 +1,117 @@
+// <copyright file="ActivityEventAttachingLogProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NET461 || NETSTANDARD2_0
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Logs
+{
+    internal class ActivityEventAttachingLogProcessor : BaseProcessor<LogRecord>
```

Comment:
I'm struggling with this name, meanwhile I don't have a good proposal...

Reply:
Ya I know it's a mouthful. Open to suggestions.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/1461
This PR suggests that span events are conceptually the same as logs.

I guess we could consider something like this?
* `TracerProvider.LogRecordToActivityEventProcessor()` (inherited from LogProcessor): Converts `LogRecord` to `ActivityEvent`, and uses `Activity.AddEvent` to attach it to the current Activity. If there there is no "current activity", this processor will simply ignore the log record.
* `LoggerProvider.ActivityEventToLogRecordProcessor(logger)` (inherited from ActivityProcessor): Converts the `ActivityEvents` objects to LogRecords.